### DISCUSSION TITLE
Streaming fast moving placemarks.

### DISCUSSION THREAD

**Discussion by irvingwa** [TerriaJS/terriajs#6674] [Q&A]
Hello everyone I am trying to figure out for the best course of action to stream geojson to the map. I have about 100k placemarks with images and labels that  gets updated (add/remove/update) around a rate of like 500 a second or so.  Is there a preferred way of doing this so the changes are done efficiently? I have been polling my datasource at a rate of every 5 seconds or so and getting the data and converting the geojson my self using the Cesium Entity API (which is consuming a lot of memory) and then I ran into this project and wanted to ask if anyone had any advice on how to best achieve this. Thank You.

**Comment by geospatialmapmashup** [comment]
dynamic pbf vector tiles from postgis database would be one option  (pg_tileserv or koopjs or any dynamic pbf vector tile package)
https://docs.terria.io/guide/connecting-to-data/catalog-type-details/mvt/
otherwise use GeoServer with POSTGIS and output WMS
https://docs.terria.io/guide/connecting-to-data/catalog-type-details/wms/
 or as tiles WMTS or TMS  (via geowebcache)  URL
https://docs.terria.io/guide/connecting-to-data/catalog-type-details/url-template-imagery/

We have about 20 million points of interest from USGS/NGA GNIS/GeoNames Data shown in Terriajs as vector tiles
and have OSM Points of Interest data as vector tiles

You can deliver vector tiles from GeoServer as well via a Community Extension
https://docs.geoserver.org/stable/en/user/extensions/vectortiles/index.html

**Reply by irvingwa** [reply]
Thank you for the response would you recommend something similar with items that are "floating" with elevation?
